### PR TITLE
Add new cop Performance/ReverseEach

### PR DESCRIPTION
**Pull request by rrosenblum**

### CODE REVIEW

File: config/enabled.yml [hunk 1]
```diff
@@ -975,6 +975,12 @@ Lint/Void:
 
 ##################### Rails ##################################
```
Comment (bbatsov) [inline config/enabled.yml]: Rails? :-)

**Comment by rrosenblum** [comment]
@bbatsov that is what I get for copying and pasting a section header. The header has been renamed to `Performance`.

File: lib/rubocop/cop/performance/reverse_each.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      #   # bad
+      #   [].reverse.each
+      #
+      #   # good
+      #   [].reverse_each
+      class ReverseEach < Cop
+        MSG = 'Use `reverse_each` instead of `reverse.each`.'
+
+        def on_send(node)
+          return unless node.children.last == :each
```
Comment (bbatsov) [inline lib/rubocop/cop/performance/reverse_each.rb]: I'm not very fond of accessing the children array, as this hides the structure of the node. We usually prefer to destructure nodes using `*`.

File: lib/rubocop/cop/performance/reverse_each.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+          return if node.children.first.nil?
+          return unless node.children.first.children.last == :reverse
```
Comment (bbatsov) [inline lib/rubocop/cop/performance/reverse_each.rb]: Same here.

File: lib/rubocop/cop/performance/reverse_each.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+          source_buffer = node.loc.expression.source_buffer
+          reverse_location = node.children.first.loc.selector.begin_pos
```
Comment (bbatsov) [inline lib/rubocop/cop/performance/reverse_each.rb]: And here. We use `node.children` pretty sparingly throughout the codebase.

**Comment by rrosenblum** [comment]
@bbatsov, sorry about the earlier usage of `node.children`. I was working on a cop for detection of parallel assignment earlier, and I was in the mindset on not using `*node`.

**Comment by bbatsov** [comment]
:+1:

**Pull request merged**